### ISSUE TITLE
DatabaseIntegrityCheck does not run on STANDBY (alwayson) databases

### ISSUE THREAD

**Issue by d3m1sc0** [olahallengren/sql-server-maintenance-solution#326]
**Description of the issue**
When running the following commands it fails.
Command: EXECUTE [dbo].[DatabaseIntegrityCheck] @databases = 'USER_DATABASES'
Error Message: Msg 976, Level 14, State 1, Procedure usp_CommandExecute, Line 194 [Batch Start Line 0]
The target database, 'dbname', is participating in an availability group and is currently not accessible for queries. Either data movement is suspended or the availability replica is not enabled for read access. To allow read-only access to this and other databases in the availability group, enable read access to one or more secondary availability replicas in the group. For more information, see the ALTER AVAILABILITY GROUP statement in SQL Server Books Online.

**SQL Server version and edition**

Microsoft SQL Server 2016 (SP2-CU10) (KB4524334) - 13.0.5492.2 (X64)   Oct  4 2019 19:14:08   Copyright (c) Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) 

**Version of the script**
 Version: 2020-01-09 23:10:28

**What command are you executing?**
EXECUTE [dbo].[DatabaseIntegrityCheck] @databases = 'USER_DATABASES'

**What output are you getting?**
Msg 976, Level 14, State 1, Procedure usp_CommandExecute, Line 194 [Batch Start Line 0]
The target database, 'dbname', is participating in an availability group and is currently not accessible for queries. Either data movement is suspended or the availability replica is not enabled for read access. To allow read-only access to this and other databases in the availability group, enable read access to one or more secondary availability replicas in the group. For more information, see the ALTER AVAILABILITY GROUP statement in SQL Server Books Online.

**Interesting finding**
the following command runs as expected:

DECLARE @command nvarchar(max)
set @command = 'DBCC CHECKDB ([dbname]) WITH NO_INFOMSGS, ALL_ERRORMSGS, DATA_PURITY'
EXECUTE (@command)

whereas the following fails with the above mentioned message:
EXECUTE @sp_executesql @command

**Comment by olahallengren** [comment]
@d3m1sc0, could you please share the complete output from the stored procedure?

**Comment by olahallengren** [comment]
I have been doing some testing on the same version and edition, and I am not able to reproduce the issue.

I can see that it is calling the stored procedure usp_CommandExecute (name modified). Could you try installing MaintenanceSolution.sql, with no modifications, and see if you still can reproduce the issue?

**Comment by d3m1sc0** [comment]
I inherited an heavily customised environment.
The maintenance solution installed was old and all the stored procedures renamed (and some customised within).
So i was trying to upgrade this to the latest.
I downloaded today again the latest (Version: 2020-01-12 16:01:32).
Installed it and run it without any changes and it run as expected (no issues).
So i will have to check our environment customisation and probably try to remove it, as it's not easily maintainable.

Thanks a lot for your time. I will close the issue.
